templates/server-ingress.yaml

```diff
@@ -1,3 +1,5 @@
+## In case of OpenShift deployment, a route will be deployed instead of ingress
+{{- if not .Values.openshift }}
```

Comment:
Is this necessary considering the default: `server.ingress.enabled: false`?

Reply:
My idea was to completely ignore server.ingress block if the target deployment platform is OpenShift. Same as ignoring server.route if deployment is on vanilla K8s. (in order to have same concept for configuring those two).
Another reason for this was that OpenShift 4 supports Ingress objects and if one is defined it will be converted to a Route (initially, I was also thinking to use ingress instead of routes, however they doesn't offer the same level of config flexibility). Without this condition in ingress object and  in case deployment is on OpenShift, the user will need to pay extra attention to disable server.ingress in values.yaml, besides enabling routes, otherwise he will get two routes created and the one converted from the ingress will not work.
But if you think it's a better idea to delegate this responsibility to the user, then we can implement this way also.